src/CodeGeneratorGo.cs

```diff
@@ -79,21 +71,55 @@ public override string ImplementationFileExtension
                     methodGroup.Name += "group";
                 }
                 ReservedFiles.Add(methodGroup.Name);
-                var methodGroupTemplate = new MethodGroupTemplate
-                {
-                    Model = methodGroup
-                };
-                await Write(methodGroupTemplate, FormatFileName(methodGroup.Name).ToLowerInvariant());
+                var methodGroupTemplate = new MethodGroupTemplate { Model = methodGroup };
+                await Write(methodGroupTemplate, FormatFileName(methodGroup.Name));
             }
 
             // Version
             var versionTemplate = new VersionTemplate { Model = codeModel };
             await Write(versionTemplate, FormatFileName("version"));
+
+            var fixedTemplates = GetFixedTemplates(codeModel);
+            foreach (var template in fixedTemplates)
+            {
+                await Write(template.Item1, FormatFileName(template.Item2));
+            }
+
+            if (codeModel.RequiresMarshallers.Any())
+            {
+                var marshallingTemplate = new Marshalling { Model = codeModel };
+                await Write(marshallingTemplate, FormatFileName("marshalling"));
+            }
+        }
+
+        /// <summary>
+        /// Returns a collection of templates to emit.
+        /// </summary>
+        /// <param name="codeModel">The code moel that the template will consume.</param>
+        /// <returns>A collection of templates, may be empty.</returns>
+        private IEnumerable<Tuple<ITemplate, string>> GetFixedTemplates(CodeModelGo codeModel)
+        {
+            var list = new List<Tuple<ITemplate, string>>
+            {
+                new Tuple<ITemplate, string>(new ResponderPolicy() { Model = codeModel }, "responder_policy"),
+                new Tuple<ITemplate, string>(new ResponseError() { Model = codeModel }, "response_error"),
+                new Tuple<ITemplate, string>(new Validation() { Model = codeModel }, "validation")

```

Comment:
You might need to add these to the ReservedFiles too

Reply:
Will fix.